sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -26,9 +26,12 @@ trait FunctionRegistry {
 
   def registerFunction(name: String, builder: FunctionBuilder): Unit
 
-  def lookupFunction(name: String, children: Seq[Expression]): Expression
-
   def caseSensitive: Boolean
+
+  def lookupFunction(
+      name: String,
+      children: Seq[Expression],
+      distinct: Boolean = false): Expression
```

Comment:
Yeah, but we couldn't tell if it's the UDF or UDAF, except the catalog itself.

Any better idea on this?